core/src/main/scala/org/bitcoins/core/gcs/GCS.scala

```diff
@@ -0,0 +1,254 @@
+package org.bitcoins.core.gcs
+
+import org.bitcoins.core.number.{UInt64, UInt8}
+import org.bouncycastle.crypto.macs.SipHash
+import org.bouncycastle.crypto.params.KeyParameter
+import scodec.bits.{BitVector, ByteVector}
+
+import scala.annotation.tailrec
+
+// TODO: Replace ByteVector with a type for keys
+/**
+  * Defines all functionality dealing with Golomb-Coded Sets as specified in
+  * [[https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#GolombCoded_Sets]]
+  */
+object GCS {
+
+  /**
+    * Given parameters and data, golomb-encodes the data as specified in
+    * [[https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#set-construction]]
```

Comment:
Same as above, when using the `@see` notation we get nice links in the Scaladoc. It's nice to follow convention:-)